anna/my-app/src/pages/moviePage/MoviePage.tsx

```diff
@@ -0,0 +1,63 @@
+import React from 'react';
+import Layout from '../../components/layout';
+import './moviePage.css';
+import MovieList from '../../components/movieList';
+import SortBox from '../../components/sortBox';
+import Container from '../../components/container';
+import MovieCard from '../../components/movieList/movie-card/MovieCard';
+import { RouteComponentProps } from "react-router-dom";
+import { MovieConnectedProps } from '.';
+import InfiniteScroll from '../../components/infiniteScroll';
+
+interface RouteParams { id: string }
+
+class MoviePage extends React.Component<RouteComponentProps<RouteParams> & MovieConnectedProps> {
+
+  componentDidUpdate(prewProps: RouteComponentProps<RouteParams> & MovieConnectedProps) {
+    const { match, movies } = this.props;
+
+    if(prewProps.movies.length === movies.length) {
+       window.scrollTo(0, 0);
+    }
+   
+    if (prewProps.match.params.id !== match.params.id) {
+      this.getMovies();
+    }
+  }
+
+  componentWillUnmount() {
+    this.props.resetMovies();
+  }
+
+  componentDidMount() {
+    console.log("mount");
+    this.getMovies();
+  }
```

Comment:
Usually we place lifecycle in the order they are called. 
componentDidMount
componentDidUpdate
componentWillUnmount